projects/storefrontlib/src/shared/components/table/table.component.html

```diff
@@ -1,40 +1,138 @@
 <ng-container *ngIf="tableData">
-  <table class="table cx-table">
-    <thead class="cx-thead-mobile">
-      <th scope="col" *ngFor="let column of columns">
-        {{ column.value }}
-      </th>
-    </thead>
-    <tbody>
-      <tr *ngFor="let row of tableData">
-        <td class="cx-code" *ngFor="let column of columns">
-          <div class="d-md-none cx-label">
-            {{ column.value }}
-          </div>
-          <label *ngIf="column.checkBox; else link">
-            <input
-              type="checkbox"
-              [checked]="row[column.key]"
-              (change)="onToggle($event, row, column.key)"
-            />
-          </label>
-          <ng-template #link>
-            <a
-              *ngIf="column.cxRoute && row; else text"
-              [routerLink]="
-                {
-                  cxRoute: column.cxRoute,
-                  params: row
-                } | cxUrl
+  <div class="desktop mode">
+    <table class="table cx-table">
+      <thead>
+        <th scope="col" *ngFor="let column of columns">
+          {{ column.value }}
+        </th>
+      </thead>
+      <tbody>
+        <tr *ngFor="let row of tableData">
+          <td class="cx-code" *ngFor="let column of columns">
+            <ng-container
+              *ngTemplateOutlet="
+                content;
+                context: { data: { row: row, column: column } }
               "
-              class="cx-value"
-            >
-              {{ row[column.key] }}
-            </a>
-          </ng-template>
-          <ng-template #text>{{ row[column.key] }}</ng-template>
-        </td>
-      </tr>
-    </tbody>
-  </table>
+            ></ng-container>
+          </td>
+        </tr>
+      </tbody>
+    </table>
+  </div>
+  <div class="mobile mode">
+    <div class="cx-row" *ngFor="let row of tableData">
+      <div
+        *ngFor="let column of columns"
+        [ngClass]="{ 'cx-action': column?.checkBox || column?.button }"
+      >
+        <span class="cx-label">{{ column.value }}: </span>
+        <ng-container
+          *ngTemplateOutlet="
+            content;
+            context: { data: { row: row, column: column } }
+          "
+        ></ng-container>
+      </div>
+    </div>
+  </div>
 </ng-container>
+
+<!-- 
+  Template for rendering checkbox input if `checkBox` property exists in column data. 
+
+  Checkbox can be used as follows in tableData object:
+  
+  key: {
+    ...,
+    selected: true
+  }
+-->
+<ng-template #checkbox let-row="row" let-column="column">
+  <ng-container *ngIf="column?.checkBox && row">
+    <input
+      type="checkbox"
+      [checked]="row[column.key]"
+      (change)="onToggle($event, row, column.key)"
+    />
+  </ng-container>
+</ng-template>
+
+<!-- 
+  Template for rendering icon button if `button` property exists in column data. 
+
+  Icon button can be used as follows in tableData object:
+  
+  key: {
```

Comment:
as I mention, it's not nice to keep metadata together with data, that properties setup could be done in html

Reply:
Done.